docs/REQUIRES_VS_IMPORTS.md

```diff
@@ -0,0 +1,470 @@
+
+- **`requires:`** - For declaring class dependencies (like "requires" in JS and "imports" in Java)
+- **`imports:`** - For dependency injection of runtime services and values
+
+## Cross-Language Design Philosophy
+
+FOAM is designed to be cross-language compatible, which means it can't easily adopt conventions from any single language. This leads to some naming that might seem counterintuitive at first:
+
+- **`requires:`** works like "requires" in JavaScript and "imports" in Java - it declares which other code/classes you want to use
+- **`imports:`** is about dependency injection - getting live runtime instances of services or objects
+
+## `requires:` - Class Dependencies
+
+### Purpose
+Declares class dependencies so you can refer to classes by their short name rather than their fully qualified name. **`requires` requires classes.**
+
+### Conceptual Model
+When you `requires: 'com.acme.Foo'`, you can refer to it as just `Foo` rather than its fully qualified name `com.acme.Foo`. This is similar to:
+- `require()` statements in JavaScript
+- `import` statements in Java
+- `#include` in C++
+
+### Syntax
+```javascript
+foam.CLASS({
+  name: 'MyClass',
+  requires: [
+    'foam.u2.DetailView',                           // Available as this.DetailView
+    'foam.dao.MDAO',                               // Available as this.MDAO
+    'foam.net.web.XMLHTTPRequest as HTTPRequest'   // Available as this.HTTPRequest
+  ]
+});
+```
+
+### How It Works
+- Classes listed in `requires` become available as factory methods
+- Accessed via `this.ClassName` (e.g., `this.DetailView`, `this.MDAO`)
+- Factory methods create new instances: `this.DetailView.create()`
+- Supports aliasing with the `as` keyword
+- **These are classes/constructors, not instances**
+
+### Examples from Codebase
+
+**Basic Usage:**
+```javascript
+// File: src/com/example/models/Transaction.js
+foam.CLASS({
+  package: 'com.example.models',
+  name: 'Transaction',
+  
+  requires: [
+    'com.example.views.TransactionFormDetailView'
+  ]
+  // Instead of: com.example.views.TransactionFormDetailView.create()
+  // You can use: this.TransactionFormDetailView.create()
+});
+```
+
+**Complex View with Multiple Dependencies:**
+```javascript
+// File: foam3/src/foam/u2/wizard/IncrementalStepWizardView.js
+foam.CLASS({
+  package: 'foam.u2.wizard',
+  name: 'IncrementalStepWizardView',
+  extends: 'foam.u2.View',
+  
+  requires: [
+    'foam.lang.Action',
+    'foam.log.LogLevel',
+    'foam.u2.detail.SectionView',
+    'foam.u2.dialog.Popup',
+    'foam.u2.dialog.SimpleActionDialog'
+  ]
+  // All classes available without full package names
+});
+```
+
+### When to Use `requires`
+- ✅ When you need to create instances of other FOAM classes
+- ✅ For view classes, DAO classes, service classes
+- ✅ When building complex UI components that compose other views
+- ✅ For factory methods and class construction
+- ✅ When you want to avoid typing full qualified class names
```

Comment:
Another difference is that when you call this.Abc.create(args), it actually calls it like this.Abc.create(args, this). ie. 'this' is added as the second argument. The second argument to create() is either a context or an object whose .__subContext__ is used. If you had created com.acme.Abc directly you would need to know to pass 'this' as the second argument, butwhen you do this.Abc.create() it is added for you. The second context argument is needed to support dependency injection / the import/export system.